src/wallet/walletdb.h

```diff
@@ -43,8 +43,16 @@ class CKeyMetadata
 {
 public:
     static const int CURRENT_VERSION=1;
+    static const int VERSION_SUPPORT_FLAGS=2;
+    static const uint8_t KEY_ORIGIN_UNSET         = 0x0000;
+    static const uint8_t KEY_ORIGIN_UNKNOWN       = 0x0001;
+    static const uint8_t KEY_ORIGIN_IMPORTED      = 0x0002;
+    static const uint8_t KEY_ORIGIN_UNENC_WALLET  = 0x0004;
+    static const uint8_t KEY_ORIGIN_ENC_WALLET    = 0x0008;
```

Comment:
I don't understand why you're using a bitfield here instead of an enumeration. Do combinations of these ever happen? In `rpcwallet.cpp` you squash it to one "enum" anyway.
(or is eg IMPORTED ENC used to signify that the key was imported while the wallet was encrypted? If so, the rpc should probably return a list of flag strings instead of one string)